internal/types/compare.go

```diff
@@ -304,9 +328,20 @@ func compareArrays(filterArr, docArr *Array) CompareResult {
 			}
 			continue
 
-		// TODO: case Document
+			// TODO this case needs to be revised and reworked to match MongoDB type sorting.
+			// Right now only tells if two documents are equal or not
 		case *Document:
-			continue
+			filterValue, err := docArr.Get(i)
+			if err != nil {
+				return Incomparable
+			}
+			filterDoc, ok := filterValue.(*Document)
+			if !ok {
+				return Incomparable
+			}
+			if compareDocuments(arrValue, filterDoc) != Equal {
+				return Incomparable
+			}
```

Comment:
try not to immediately return a `Incomparable`, because sometimes the value that satisfies the filter might be at the end of the array. Use `continue` instead of returns.

Reply:
Yeah, it's a silly mistake, thanks for noticing that